src/features/code-input-and-control/code-input-and-control.tsx

```diff
@@ -0,0 +1,20 @@
+import CodeInputController from "../code-input-controller/code-input-controller";
+import CodeInput from "../code-input/code-input.module";
+
+interface CodeInputAndControllerProperties {
+  editor_id: number
+}
+
+export default function CodeInputAndController(props: CodeInputAndControllerProperties) {
+
+  return (
+    <div className="code-editor-and-controls">
```

Comment:
Den State (den eingegebenen Code) der Components kann man natürlich global über den store sharen, da dieser aber bislang nur in den beiden hier aufgeführten Componenten verwendet wird, würde ich den in der Praxis aus dem globalen store entfernen und nur hier lokal halten, also sowas wie (pseudocode)

```
const [code,setCode] = useState<string>('')

return (
    <div className="code-editor-and-controls">
      <div className="code-editor-controls">
        <CodeInputController code={code} />
      </div>
      <div className="code-editor-wrapper">
        <CodeInput code={code} setCode={(text) => setCode(text)} />
      </div>
    </div>
  )
```

Dann kannst du die ganze Editor ID sparen, die von ganz oben aus der App mitschleifen musst. Das macht die Komponente flexibler, da sie nun einfach irgendwo beliebig erstellt werden kann, ohne, dass man sich gedanken darüber machen muss, wie man die ids verwaltet.